### PR TITLE
feat(android): signInWithBrowser - get authorization code

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

This feature is split into two parts:
* get authorization code
* fetch token by authorization code

This PR is part one: get authorization code.

Todo: Refactor Completion

## Linear Issue Reference

* LOG-1098

## Testing

Success:
image
Cancel:
image

### CODE REVIEW

**Review by gao-sun — APPROVED** [review]

File: android-sdk/android/src/main/kotlin/io/logto/sdk/android/auth/session/SignInSessionManager.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package io.logto.sdk.android.auth.session
+
+import android.annotation.SuppressLint
+import android.net.Uri
+
+object SignInSessionManager {
```
Comment (simeng-li) [inline android-sdk/android/src/main/kotlin/io/logto/sdk/android/auth/session/SignInSessionManager.kt:6]: Actually do we need this extra SignInSessionManager layer? Is it possible to create a private `SignInSession` instance within the `LogtoClient` itself?

File: android-sdk/android/src/main/kotlin/io/logto/sdk/android/LogtoClient.kt [hunk 2]
```diff
@@ -30,6 +33,28 @@ open class LogtoClient(
...
+        context: Activity,
+        redirectUri: String,
+        completion: Completion<String>,
+    ) = getOidcConfig { getOidcConfigException, oidcConfig ->
+        getOidcConfigException?.let {
+            completion.onComplete(getOidcConfigException, null)
+            return@getOidcConfig
+        }
+
+        val signInSession = SignInSession(
```
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply android-sdk/android/src/main/kotlin/io/logto/sdk/android/LogtoClient.kt:47]: Will be auto released by JVM

File: android-sdk/android/src/main/kotlin/io/logto/sdk/android/auth/session/SignInSessionManager.kt (hunk 1 above)
Comment (xiaoyijun) [reply android-sdk/android/src/main/kotlin/io/logto/sdk/android/auth/session/SignInSessionManager.kt:6]: SignInSessionManager is a global singleton. `LogtoClient` is not global.
We need to use SignInSessionManager to hold the auth context.
It's equivalent to the `AuthManager` in the old code.

File: android-sample/app/src/main/java/io/logto/demo/fragment/LoginFragment.kt [hunk 3]
```diff
@@ -7,8 +7,21 @@ import android.view.View
...
 import android.widget.Button
 import io.logto.demo.R
+import io.logto.sdk.android.LogtoClient
+import io.logto.sdk.android.type.LogtoConfig
 
 class LoginFragment : Fragment() {
+
+    val logtoConfig = LogtoConfig(
+        endpoint = "https://logto.dev",
+        clientId = "z4skkM1Z8LLVSl1JCmVZO",
```
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply android-sample/app/src/main/java/io/logto/demo/fragment/LoginFragment.kt:17]: Good call.
LOG-1480: Use Specific Client IDs For Our Demo App

**Pull request merged**